File: 0349/solution.go
package main

func intersection(nums1 []int, nums2 []int) []int {
	set1 := make(map[int]bool)
	set2 := make(map[int]bool)
	for _, num := range nums1 {
		set1[num] = true
	}
	for _, num := range nums2 {
		set2[num] = true
	}
	res := make([]int, 0)
	for key := range set1 {
		if set2[key] {
			res = append(res, key)
		}
	}
	return res
}

File: 0349/solution.ts
function intersection(nums1: number[], nums2: number[]): number[] {
  const set1 = new Set(nums1);
  const set2 = new Set(nums2);
  const res: number[] = [];
  for (const num of set1) {
    if (set2.has(num)) {
      res.push(num);
    }
  }
  return res;
}
